trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/testing/ManagedS3MockContainer.java

```diff
@@ -63,11 +64,13 @@ public class ManagedS3MockContainer
     @BindingAnnotation
     public @interface ForS3MockContainer {}
 
+    public record ContainerHost(String host, Integer port) {}
```

Comment:
Trino-ism - we always add a canonical constructor that checks for `null` arguments. This has a lot of benefits. We get a null safety everywhere at a low cost.

Reply:
Yup this was my bad, since this record was only used for testing it slipped my mind - thanks!